argparse/__init__.py

```diff
@@ -115,31 +118,34 @@ def __init__(self,
 
     def add_argument(self, *args, **kwargs):
         result = ap.ArgumentParser.add_argument(self, *args, **kwargs)
+        # to avoid dublicate ids when there's a positional and an optional param with the same dest
+        argument_names = [arg.dest for arg in self.argument_list]
+        if result.dest in argument_names:
+            result.dest = '_' + result.dest
         self.argument_list.append(result)
 
     def parse_args(self, *args, **kwargs):
 
         arg2cwl_parser = get_arg2cwl_parser()
-
         if '--generate_cwl_tool' in sys.argv:
             arg2cwl_parser.add_argument('--generate_cwl_tool', action='store_true')
             arg2cwl_args = arg2cwl_parser.parse_args(*args, **kwargs)
-
+            print('sys argv: ', sys.argv)
```

Comment:
As part of the cleanup, it's perhaps worth start using python's `logging` module instead of `print()`: http://victorlin.me/posts/2012/08/26/good-logging-practice-in-python